content/tutorials/use-specialized-payment-types/create-custom-transactors.md

```diff
@@ -0,0 +1,379 @@
+    // check the starting balance because we want to allow dipping into the
+    // reserve to pay fees.
+    {
+        STAmount const reserve{
+            view().fees().accountReserve(sle->getFieldU32(sfOwnerCount) + 1)};
+
+        if (mPriorBalance < reserve)
+            return tecINSUFFICIENT_RESERVE;
+    }
+
+    // Note that we use the value from the sequence or ticket as the
+    // Check sequence.  For more explanation see comments in SeqProxy.h.
+    std::uint32_t const seq = ctx_.tx.getSeqProxy().value();
+    Keylet const checkKeylet = keylet::check(account_, seq);
+    auto sleCheck = std::make_shared<SLE>(checkKeylet);
+
+    sleCheck->setAccountID(sfAccount, account_);
+    AccountID const dstAccountId = ctx_.tx[sfDestination];
+    sleCheck->setAccountID(sfDestination, dstAccountId);
+    sleCheck->setFieldU32(sfSequence, seq);
+    sleCheck->setFieldAmount(sfSendMax, ctx_.tx[sfSendMax]);
+    if (auto const srcTag = ctx_.tx[~sfSourceTag])
+        sleCheck->setFieldU32(sfSourceTag, *srcTag);
+    if (auto const dstTag = ctx_.tx[~sfDestinationTag])
+        sleCheck->setFieldU32(sfDestinationTag, *dstTag);
+    if (auto const invoiceId = ctx_.tx[~sfInvoiceID])
+        sleCheck->setFieldH256(sfInvoiceID, *invoiceId);
+    if (auto const expiry = ctx_.tx[~sfExpiration])
+        sleCheck->setFieldU32(sfExpiration, *expiry);
+
+    view().insert(sleCheck);
+
+    auto viewJ = ctx_.app.journal("View");
+    // If it's not a self-send (and it shouldn't be), add Check to the
+    // destination's owner directory.
+    if (dstAccountId != account_)
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(dstAccountId),
+            checkKeylet,
+            describeOwnerDir(dstAccountId));
+
+        JLOG(j_.trace()) << "Adding Check to destination directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfDestinationNode, *page);
+    }
+
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(account_),
+            checkKeylet,
+            describeOwnerDir(account_));
+
+        JLOG(j_.trace()) << "Adding Check to owner directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfOwnerNode, *page);
+    }
+    // If we succeeded, the new entry counts against the creator's reserve.
+    adjustOwnerCount(view(), sle, 1, viewJ);
+    return tesSUCCESS;
+}
+```
+
+
+## Additional Functions
+
+
+### `calculateBaseFee`
+
+The `calculateBaseFee` function calculates the XRP fee to run a transaction. The fee is usually inherited from the base transactor, but transactors such as `EscrowFinish` require additional fees.
```

Comment:
This phrasing, I think, hides the intent here. Here's an attempt to make it more obvious:

```suggestion
Most transactions inherit the default [reference transaction cost](transaction-cost.html). However, if your transactor needs to define a non-standard transaction cost, you can replace the transactor's `calculateBaseFee` method with a custom one. For example, the following example shows how `EscrowFinish` transactions charge an additional cost on conditional escrows based on the size of the fulfillment:
```